be/src/io/shared_buffered_input_stream.h

```diff
@@ -79,7 +79,16 @@ class SharedBufferedInputStream : public SeekableInputStream {
 
     Status set_io_ranges(const std::vector<IORange>& ranges, bool coalesce_lazy_column = true);
     void release_to_offset(int64_t offset);
-    void release();
+    void release() override;
+
+    // If the range of IO Coalease is limited between a single column in shared data,
+    // we can release the buffer in advance when column_iterator get eos.
+    // If it is the entire file (lake_small_segment_file_threshold_size > file_size), it cannot be released in advance.
+    // return false only ranges of shared_buffer is the whole file.
+    bool can_release() override {
```

Comment:
will this be too complicated for the releasing? how about each column iterator reset its shared_ptr when reaches eos. the last one who releases its shared_ptr, the sharedbufferinputstream will be released.

Reply:
the twisted point here is, the sharing is a pure caller's action, and now it moves to the callee's duty to know whether the sharing still holds and it can do the correct action based on caller's sharing state. That can't be right as always.

What I can think of a solution may look like: wrapping the sharedbufferinputstream with a reference counting, when doing the init, the reference count increases, when any iter reaches eos, do a decrease. when the reference down to 0, do whatever you want to clean the resource.